src/Util/GeometryUtil/GeometryUtil.js

```diff
@@ -62,58 +58,77 @@ class GeometryUtil {
         geometry: line
       });
 
-    // Convert the input features to turf.js/GeoJSON features while
+    // Convert the input line features to turf.js/GeoJSON features while
     // reprojecting them to the internal turf.js projection 'EPSG:4326'.
-    const turfPolygon = geoJsonFormat.writeFeatureObject(polygonFeat);
     const turfLine = geoJsonFormat.writeFeatureObject(lineFeat);
-
-    // Union both geometries to a set of MultiLineStrings.
-    const unionGeom = union(polygonToLine(turfPolygon), turfLine);
-
-    // Buffer the input polygon to take great circle (un-)precision
-    // into account.
-    const bufferedTurfPolygon = buffer(turfPolygon, tolerance, {
-      units: 'meters'
-    });
-
-    let filteredSegments = [];
-
-    // Iterate over each segment and remove any segment that is not covered by
-    // the (buffered) input polygon.
-    segmentEach(unionGeom, (currentSegment, featureIndex, multiFeatureIndex) => {
-      const segmentCenter = centroid(currentSegment);
-      const isSegmentInPolygon = booleanPointInPolygon(segmentCenter, bufferedTurfPolygon);
-
-      if (isSegmentInPolygon) {
-        if (!filteredSegments[multiFeatureIndex]) {
-          filteredSegments[multiFeatureIndex] = [];
-        }
-
-        if (filteredSegments[multiFeatureIndex].length === 0) {
-          filteredSegments[multiFeatureIndex].push(
-            getCoords(currentSegment)[0],
-            getCoords(currentSegment)[1]
-          );
-        } else {
-          filteredSegments[multiFeatureIndex].push(
-            getCoords(currentSegment)[1]
-          );
+    // This lists all the polygons in the feature and splits the Multi polygons into an array of polygons.
+    const geometries = GeometryUtil.separateGeometries(polygonFeat.getGeometry());
+    // the array containing all the splited features 
+    let allSplitedPolygons = [];
+    // iterates over each polygon and splits it
+    geometries.forEach(geometry => {
+      // Convert the polygon to turf.js/GeoJSON geometry while
+      // reprojecting them to the internal turf.js projection 'EPSG:4326'.
+      const turfPolygon = geoJsonFormat.writeGeometryObject(geometry);
+      const turfPolygonCoordinates = turfPolygon.coordinates;
+      // ouline lines of the given polygon
+      const outer = lineString(turfPolygonCoordinates[0]);
```

Comment:
Can we surely assume, the first index contains the outer polygon at all times? Maybe we could enhance this by checking the draw direction of the linestring with `booleanClockwise()`.

Reply:
In the <a href="http://wiki.geojson.org/GeoJSON_draft_version_6">GeoJSON specifications</a>  it states that:
"For type "Polygon", each element in the coordinates array is a coordinates array as described for type "LineString". Furthermore, each LineString in the coordinates array must be a LinearRing. For Polygons with multiple LinearRings, the <b>first must be the exterior ring and any others must be interior rings or holes</b>." 
So I think it is not necessary.

Reply:
Top!